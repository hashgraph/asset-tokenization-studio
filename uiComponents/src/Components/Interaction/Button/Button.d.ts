import type { ButtonProps as ChakraButtonProps } from "@chakra-ui/button";
import type { SystemStyleObject, ComponentWithAs } from "@chakra-ui/system";
import type { BaseSingleStyleConfiguration } from "@Theme/Components/BaseSingleStyleConfiguration";
export type ButtonVariantProps = {
    status: ButtonProps["colorScheme"];
};
export interface ButtonThemeConfiguration extends Omit<BaseSingleStyleConfiguration, "variants"> {
    baseStyle: Partial<SystemStyleObject>;
    variants?: {
        [key: string]: ((props: ButtonVariantProps) => Partial<SystemStyleObject>) | Partial<SystemStyleObject>;
    };
}
export interface ButtonProps extends ChakraButtonProps {
    status?: string;
}
export declare const Button: ComponentWithAs<"button", ButtonProps>;
