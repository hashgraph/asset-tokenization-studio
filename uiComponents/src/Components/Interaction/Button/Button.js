var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from "react";
import { Button as ChakraButton } from "@chakra-ui/button";
import { useStyleConfig } from "@chakra-ui/react";
import { forwardRef } from "@chakra-ui/system";
import { Spinner } from "../../Indicators/Spinner/Spinner";
import _merge from "lodash/merge";
export var Button = forwardRef(function (_a, ref) {
    var children = _a.children, sx = _a.sx, status = _a.status, props = __rest(_a, ["children", "sx", "status"]);
    var themeStyles = useStyleConfig("Button", __assign(__assign({}, props), { status: status }));
    var loadingText = typeof children === "string" && !props.loadingText
        ? children
        : props.loadingText;
    var styles = React.useMemo(function () { return _merge(themeStyles, sx); }, [themeStyles, sx]);
    return (React.createElement(ChakraButton, __assign({ ref: ref, loadingText: loadingText, spinner: React.createElement(Spinner, { size: "xxs" }) }, props, { sx: styles }), children));
});
