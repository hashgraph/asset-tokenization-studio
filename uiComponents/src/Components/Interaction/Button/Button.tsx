import React from "react";
import type { ButtonProps as ChakraButtonProps } from "@chakra-ui/button";
import type { SystemStyleObject, ComponentWithAs } from "@chakra-ui/system";
import { Button as ChakraButton } from "@chakra-ui/button";
import { useStyleConfig } from "@chakra-ui/react";
import { forwardRef } from "@chakra-ui/system";
import type { BaseSingleStyleConfiguration } from "@Theme/Components/BaseSingleStyleConfiguration";
import { Spinner } from "../../Indicators/Spinner/Spinner";
import _merge from "lodash/merge";

export type ButtonVariantProps = {
  status: ButtonProps["colorScheme"];
};

export interface ButtonThemeConfiguration
  extends Omit<BaseSingleStyleConfiguration, "variants"> {
  baseStyle: Partial<SystemStyleObject>;
  variants?: {
    [key: string]:
      | ((props: ButtonVariantProps) => Partial<SystemStyleObject>)
      | Partial<SystemStyleObject>;
  };
}

export interface ButtonProps extends ChakraButtonProps {
  status?: string;
}

export const Button: ComponentWithAs<"button", ButtonProps> = forwardRef<
  ButtonProps,
  "button"
>(({ children, sx, status, ...props }: ButtonProps, ref) => {
  const themeStyles = useStyleConfig("Button", {
    ...props,
    status,
  });
  const loadingText =
    typeof children === "string" && !props.loadingText
      ? children
      : props.loadingText;

  const styles = React.useMemo(
    () => _merge(themeStyles, sx),
    [themeStyles, sx]
  );

  return (
    <ChakraButton
      ref={ref}
      loadingText={loadingText}
      spinner={<Spinner size="xxs" />}
      {...props}
      sx={styles}
    >
      {children}
    </ChakraButton>
  );
});
