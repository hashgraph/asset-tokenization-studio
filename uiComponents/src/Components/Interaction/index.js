export * from "./Button";
export * from "./CardButton";
export * from "./ClipboardButton";
export * from "./CloseButton";
export * from "./HoverableContent";
export * from "./IconButton";
export * from "./Link";
