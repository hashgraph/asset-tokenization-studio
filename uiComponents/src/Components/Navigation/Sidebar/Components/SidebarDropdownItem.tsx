import type { PhosphorIconProps } from "@/Components/Foundations/PhosphorIcon";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import { Text } from "@/Components/Foundations/Text";
import type { MenuItemProps as ChakraMenuItemProps } from "@chakra-ui/react";
import { MenuItem as ChakraMenuItem } from "@chakra-ui/react";
import React from "react";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";
import type { SidebarDropdownThemeStyle } from "./SidebarItemWithDropdown";

export interface SidebarDropdownItemProps extends ChakraMenuItemProps {
  label: string;
  icon?: PhosphorIconProps["as"];
  isDisabled?: boolean;
  isActive?: boolean;
}
export const SidebarDropdownItem = ({
  label,
  icon,
  isDisabled,
  isActive,
  ...props
}: SidebarDropdownItemProps) => {
  const styles = useChakraMultiStyleConfig("SidebarDropdown", {
    isActive,
    isDisabled,
  }) as SidebarDropdownThemeStyle;

  return (
    <ChakraMenuItem
      as="button"
      data-testid={`sidebar-dropdown-item-${label}`}
      disabled={isDisabled}
      {...props}
      sx={styles.itemContainer}
    >
      {icon && (
        <PhosphorIcon
          data-testid="sidebar-dropdown-item-icon"
          as={icon}
          size="xs"
          weight={styles.itemIconWeight}
        />
      )}
      <Text sx={styles.itemLabel}>{label}</Text>
    </ChakraMenuItem>
  );
};
