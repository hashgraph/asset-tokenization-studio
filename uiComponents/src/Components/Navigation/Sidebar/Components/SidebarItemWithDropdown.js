import { Box as ChakraBox, Menu as ChakraMenu, MenuButton as ChakraMenuButton, MenuList as ChakraMenuList, Stack as ChakraStack, useDisclosure as useChakraDisclosure, } from "@chakra-ui/react";
import React from "react";
import { SidebarItem } from "./SidebarItem";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";
export var sidebarDropdownPartsList = [
    "header",
    "wrapper",
    "container",
    "itemIconWeight",
    "itemLabel",
    "itemContainer",
];
export var SidebarItemWithDropdown = function (_a) {
    var label = _a.label, header = _a.header, icon = _a.icon, children = _a.children, isActive = _a.isActive, isDisabled = _a.isDisabled;
    var styles = useChakraMultiStyleConfig("SidebarDropdown", {
        hasHeader: !!header
    });
    var _b = useChakraDisclosure(), isOpen = _b.isOpen, onOpen = _b.onOpen, onClose = _b.onClose;
    return (React.createElement(ChakraMenu, { placement: "right-start", gutter: 0, isOpen: isOpen },
        React.createElement(ChakraMenuButton, { onMouseEnter: onOpen, onMouseLeave: onClose, as: SidebarItem, label: label, icon: icon, isActive: isActive, isDisabled: isDisabled }),
        React.createElement(ChakraMenuList, { position: "relative", sx: styles.wrapper, onMouseEnter: onOpen, onMouseLeave: onClose },
            header && (React.createElement(ChakraBox, { "data-testid": "sidebar-dropdown-header", sx: styles.header }, header)),
            React.createElement(ChakraStack, { spacing: 1, sx: styles.container }, children))));
};
