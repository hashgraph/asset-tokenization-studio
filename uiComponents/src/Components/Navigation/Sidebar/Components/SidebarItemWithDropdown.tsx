import {
  Box as ChakraBox,
  Menu as ChakraMenu,
  MenuButton as ChakraMenuButton,
  MenuList as ChakraMenuList,
  Stack as ChakraStack,
  useDisclosure as useChakraDisclosure,
} from "@chakra-ui/react";
import React from "react";
import type { SidebarItemProps } from "./SidebarItem";
import { SidebarItem } from "./SidebarItem";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";

import type { SystemStyleObject as ChakraSystemStyleObject } from "@chakra-ui/system";
import type {
  BaseMultiStyleConfiguration,
  PartsStyleInterpolation,
} from "@/Theme/Components/BaseMultiStyleConfiguration";
import type { Weight } from "@/Components/Foundations/PhosphorIcon";

export interface SidebarItemWithDropdownProps extends SidebarItemProps {
  header?: React.ReactElement;
  children: React.ReactElement | React.ReactElement[];
}

export const sidebarDropdownPartsList: Array<
  | "header"
  | "wrapper"
  | "container"
  | "itemIconWeight"
  | "itemLabel"
  | "itemContainer"
> = [
  "header",
  "wrapper",
  "container",
  "itemIconWeight",
  "itemLabel",
  "itemContainer",
];

type Parts = typeof sidebarDropdownPartsList;

export type SidebarDropdownConfigProps = {
  isDisabled?: boolean;
  hasHeader?: boolean;
  isActive?: boolean;
};

export interface SidebarDropdownThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  baseStyle:
    | (({
        hasHeader,
        isDisabled,
        isActive,
      }: SidebarDropdownConfigProps) => Partial<SidebarDropdownThemeStyle>)
    | PartsStyleInterpolation<Parts>;
}

export type SidebarDropdownThemeStyle = Record<
  Parts[number],
  ChakraSystemStyleObject
> & {
  itemIconWeight: Weight;
};

export const SidebarItemWithDropdown = ({
  label,
  header,
  icon,
  children,
  isActive,
  isDisabled,
}: SidebarItemWithDropdownProps) => {
  const styles = useChakraMultiStyleConfig("SidebarDropdown", {
    hasHeader: !!header,
  }) as SidebarDropdownThemeStyle;
  const { isOpen, onOpen, onClose } = useChakraDisclosure();

  return (
    <ChakraMenu placement="right-start" gutter={0} isOpen={isOpen}>
      <ChakraMenuButton
        onMouseEnter={onOpen}
        onMouseLeave={onClose}
        as={SidebarItem}
        label={label}
        icon={icon}
        isActive={isActive}
        isDisabled={isDisabled}
      />
      <ChakraMenuList
        position="relative"
        sx={styles.wrapper}
        onMouseEnter={onOpen}
        onMouseLeave={onClose}
      >
        {header && (
          <ChakraBox data-testid="sidebar-dropdown-header" sx={styles.header}>
            {header}
          </ChakraBox>
        )}
        <ChakraStack spacing={1} sx={styles.container}>
          {children}
        </ChakraStack>
      </ChakraMenuList>
    </ChakraMenu>
  );
};
