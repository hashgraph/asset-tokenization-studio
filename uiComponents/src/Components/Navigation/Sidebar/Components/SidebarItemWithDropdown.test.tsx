import { render } from "@/test-utils";
import { Cookie, UserPlus } from "@phosphor-icons/react";
import React from "react";
import type { SidebarItemWithDropdownProps } from "./SidebarItemWithDropdown";
import { SidebarItemWithDropdown } from "./SidebarItemWithDropdown";
import { fireEvent, waitFor } from "@testing-library/react";
import { Menu } from "@chakra-ui/react";
import { SidebarDropdownItem } from "./SidebarDropdownItem";
import { Button } from "@Components/Interaction/Button";

describe(`<SidebarItemWithDropdown />`, () => {
  const defaultProps: SidebarItemWithDropdownProps = {
    label: "Item",
    icon: Cookie,
    children: (
      <>
        <SidebarDropdownItem label="Page 1" icon={UserPlus} />
        <SidebarDropdownItem label="Page 2" icon={UserPlus} isActive={true} />
      </>
    ),
  };
  const factoryComponent = (
    props: SidebarItemWithDropdownProps = defaultProps
  ) =>
    render(
      <Menu>
        <SidebarItemWithDropdown {...props} />
      </Menu>
    );

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
  });

  test("renders correctly when hovered", async () => {
    const component = factoryComponent();
    fireEvent.mouseEnter(
      component.getByTestId(`sidebar-item-${defaultProps.label}`)
    );

    await waitFor(() => {
      expect(component.asFragment()).toMatchSnapshot("Hovered");
    });
  });

  test("renders correctly with header", () => {
    const component = factoryComponent({
      ...defaultProps,
      header: (
        <Button variant="secondary" size="md">
          Main action
        </Button>
      ),
    });

    expect(component.getByTestId("sidebar-dropdown-header")).toHaveTextContent(
      "Main action"
    );
    expect(component.asFragment()).toMatchSnapshot("WithHeader");
  });
});
