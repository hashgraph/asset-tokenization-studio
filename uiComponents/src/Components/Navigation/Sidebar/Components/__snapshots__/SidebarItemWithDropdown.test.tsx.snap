// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarItemWithDropdown /> renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-controls="menu-list-:r4:"
    aria-expanded="false"
    aria-haspopup="menu"
    class="chakra-menu__menu-button css-df6r3v"
    data-testid="sidebar-item-Item"
    id="menu-button-:r4:"
  >
    <button
      class="css-1rb62th"
    >
      <svg
        class="chakra-icon css-1qzovvq"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M164.49,163.51a12,12,0,1,1-17,0A12,12,0,0,1,164.49,163.51Zm-81-8a12,12,0,1,0,17,0A12,12,0,0,0,83.51,155.51Zm9-39a12,12,0,1,0-17,0A12,12,0,0,0,92.49,116.49Zm48-1a12,12,0,1,0,0,17A12,12,0,0,0,140.49,115.51ZM232,128A104,104,0,1,1,128,24a8,8,0,0,1,8,8,40,40,0,0,0,40,40,8,8,0,0,1,8,8,40,40,0,0,0,40,40A8,8,0,0,1,232,128Zm-16.31,7.39A56.13,56.13,0,0,1,168.5,87.5a56.13,56.13,0,0,1-47.89-47.19,88,88,0,1,0,95.08,95.08Z"
        />
      </svg>
    </button>
    <p
      class="css-owi0ee"
    >
      Item 
    </p>
  </div>
  <div
    class="css-r6z5ec"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-1ayccjv"
      id="menu-list-:r4:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="chakra-stack css-9jw3b1"
      >
        <button
          class="chakra-menu__menuitem css-qqyk6q"
          data-index="0"
          data-testid="sidebar-dropdown-item-Page 1"
          id="menu-list-:r4:-menuitem-:r5:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            data-testid="sidebar-dropdown-item-icon"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256,136a8,8,0,0,1-8,8H232v16a8,8,0,0,1-16,0V144H200a8,8,0,0,1,0-16h16V112a8,8,0,0,1,16,0v16h16A8,8,0,0,1,256,136Zm-57.87,58.85a8,8,0,0,1-12.26,10.3C165.75,181.19,138.09,168,108,168s-57.75,13.19-77.87,37.15a8,8,0,0,1-12.25-10.3c14.94-17.78,33.52-30.41,54.17-37.17a68,68,0,1,1,71.9,0C164.6,164.44,183.18,177.07,198.13,194.85ZM108,152a52,52,0,1,0-52-52A52.06,52.06,0,0,0,108,152Z"
            />
          </svg>
          <p
            class="css-pyxwkt"
          >
            Page 1
          </p>
        </button>
        <button
          class="chakra-menu__menuitem css-xe6699"
          data-index="1"
          data-testid="sidebar-dropdown-item-Page 2"
          id="menu-list-:r4:-menuitem-:r6:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            data-testid="sidebar-dropdown-item-icon"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256,136a8,8,0,0,1-8,8H232v16a8,8,0,0,1-16,0V144H200a8,8,0,0,1,0-16h16V112a8,8,0,0,1,16,0v16h16A8,8,0,0,1,256,136ZM144,157.68a68,68,0,1,0-71.9,0c-20.65,6.76-39.23,19.39-54.17,37.17A8,8,0,0,0,24,208H192a8,8,0,0,0,6.13-13.15C183.18,177.07,164.6,164.44,144,157.68Z"
            />
          </svg>
          <p
            class="css-1n40qdq"
          >
            Page 2
          </p>
        </button>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<SidebarItemWithDropdown /> renders correctly when hovered: Hovered 1`] = `
<DocumentFragment>
  <div
    aria-controls="menu-list-:rb:"
    aria-expanded="true"
    aria-haspopup="menu"
    class="chakra-menu__menu-button css-df6r3v"
    data-active=""
    data-testid="sidebar-item-Item"
    id="menu-button-:rb:"
  >
    <button
      class="css-ek0iur"
    >
      <svg
        class="chakra-icon css-1qzovvq"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M164.49,163.51a12,12,0,1,1-17,0A12,12,0,0,1,164.49,163.51Zm-81-8a12,12,0,1,0,17,0A12,12,0,0,0,83.51,155.51Zm9-39a12,12,0,1,0-17,0A12,12,0,0,0,92.49,116.49Zm48-1a12,12,0,1,0,0,17A12,12,0,0,0,140.49,115.51ZM232,128A104,104,0,1,1,128,24a8,8,0,0,1,8,8,40,40,0,0,0,40,40,8,8,0,0,1,8,8,40,40,0,0,0,40,40A8,8,0,0,1,232,128Zm-16.31,7.39A56.13,56.13,0,0,1,168.5,87.5a56.13,56.13,0,0,1-47.89-47.19,88,88,0,1,0,95.08,95.08Z"
        />
      </svg>
    </button>
    <p
      class="css-owi0ee"
    >
      Item 
    </p>
  </div>
  <div
    class="css-r6z5ec"
    data-popper-escaped=""
    data-popper-placement="right-start"
    data-popper-reference-hidden=""
    style="visibility: visible; position: absolute; min-width: max-content; inset: 0 auto auto 0; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 0px); --popper-transform-origin: left top;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-1ayccjv"
      id="menu-list-:rb:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="chakra-stack css-9jw3b1"
      >
        <button
          class="chakra-menu__menuitem css-qqyk6q"
          data-index="0"
          data-testid="sidebar-dropdown-item-Page 1"
          id="menu-list-:rb:-menuitem-:rc:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            data-testid="sidebar-dropdown-item-icon"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256,136a8,8,0,0,1-8,8H232v16a8,8,0,0,1-16,0V144H200a8,8,0,0,1,0-16h16V112a8,8,0,0,1,16,0v16h16A8,8,0,0,1,256,136Zm-57.87,58.85a8,8,0,0,1-12.26,10.3C165.75,181.19,138.09,168,108,168s-57.75,13.19-77.87,37.15a8,8,0,0,1-12.25-10.3c14.94-17.78,33.52-30.41,54.17-37.17a68,68,0,1,1,71.9,0C164.6,164.44,183.18,177.07,198.13,194.85ZM108,152a52,52,0,1,0-52-52A52.06,52.06,0,0,0,108,152Z"
            />
          </svg>
          <p
            class="css-pyxwkt"
          >
            Page 1
          </p>
        </button>
        <button
          class="chakra-menu__menuitem css-xe6699"
          data-index="1"
          data-testid="sidebar-dropdown-item-Page 2"
          id="menu-list-:rb:-menuitem-:rd:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            data-testid="sidebar-dropdown-item-icon"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256,136a8,8,0,0,1-8,8H232v16a8,8,0,0,1-16,0V144H200a8,8,0,0,1,0-16h16V112a8,8,0,0,1,16,0v16h16A8,8,0,0,1,256,136ZM144,157.68a68,68,0,1,0-71.9,0c-20.65,6.76-39.23,19.39-54.17,37.17A8,8,0,0,0,24,208H192a8,8,0,0,0,6.13-13.15C183.18,177.07,164.6,164.44,144,157.68Z"
            />
          </svg>
          <p
            class="css-1n40qdq"
          >
            Page 2
          </p>
        </button>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<SidebarItemWithDropdown /> renders correctly with header: WithHeader 1`] = `
<DocumentFragment>
  <div
    aria-controls="menu-list-:ri:"
    aria-expanded="false"
    aria-haspopup="menu"
    class="chakra-menu__menu-button css-df6r3v"
    data-testid="sidebar-item-Item"
    id="menu-button-:ri:"
  >
    <button
      class="css-1rb62th"
    >
      <svg
        class="chakra-icon css-1qzovvq"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M164.49,163.51a12,12,0,1,1-17,0A12,12,0,0,1,164.49,163.51Zm-81-8a12,12,0,1,0,17,0A12,12,0,0,0,83.51,155.51Zm9-39a12,12,0,1,0-17,0A12,12,0,0,0,92.49,116.49Zm48-1a12,12,0,1,0,0,17A12,12,0,0,0,140.49,115.51ZM232,128A104,104,0,1,1,128,24a8,8,0,0,1,8,8,40,40,0,0,0,40,40,8,8,0,0,1,8,8,40,40,0,0,0,40,40A8,8,0,0,1,232,128Zm-16.31,7.39A56.13,56.13,0,0,1,168.5,87.5a56.13,56.13,0,0,1-47.89-47.19,88,88,0,1,0,95.08,95.08Z"
        />
      </svg>
    </button>
    <p
      class="css-owi0ee"
    >
      Item 
    </p>
  </div>
  <div
    class="css-r6z5ec"
    style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-j55j5w"
      id="menu-list-:ri:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="css-kfy7fn"
        data-testid="sidebar-dropdown-header"
      >
        <button
          class="chakra-button css-trpjf1"
          type="button"
        >
          Main action
        </button>
      </div>
      <div
        class="chakra-stack css-9jw3b1"
      >
        <button
          class="chakra-menu__menuitem css-qqyk6q"
          data-index="0"
          data-testid="sidebar-dropdown-item-Page 1"
          id="menu-list-:ri:-menuitem-:rj:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            data-testid="sidebar-dropdown-item-icon"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256,136a8,8,0,0,1-8,8H232v16a8,8,0,0,1-16,0V144H200a8,8,0,0,1,0-16h16V112a8,8,0,0,1,16,0v16h16A8,8,0,0,1,256,136Zm-57.87,58.85a8,8,0,0,1-12.26,10.3C165.75,181.19,138.09,168,108,168s-57.75,13.19-77.87,37.15a8,8,0,0,1-12.25-10.3c14.94-17.78,33.52-30.41,54.17-37.17a68,68,0,1,1,71.9,0C164.6,164.44,183.18,177.07,198.13,194.85ZM108,152a52,52,0,1,0-52-52A52.06,52.06,0,0,0,108,152Z"
            />
          </svg>
          <p
            class="css-pyxwkt"
          >
            Page 1
          </p>
        </button>
        <button
          class="chakra-menu__menuitem css-xe6699"
          data-index="1"
          data-testid="sidebar-dropdown-item-Page 2"
          id="menu-list-:ri:-menuitem-:rk:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            data-testid="sidebar-dropdown-item-icon"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256,136a8,8,0,0,1-8,8H232v16a8,8,0,0,1-16,0V144H200a8,8,0,0,1,0-16h16V112a8,8,0,0,1,16,0v16h16A8,8,0,0,1,256,136ZM144,157.68a68,68,0,1,0-71.9,0c-20.65,6.76-39.23,19.39-54.17,37.17A8,8,0,0,0,24,208H192a8,8,0,0,0,6.13-13.15C183.18,177.07,164.6,164.44,144,157.68Z"
            />
          </svg>
          <p
            class="css-1n40qdq"
          >
            Page 2
          </p>
        </button>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
